csrc/hkdf.cpp

```diff
@@ -0,0 +1,114 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+#include "buffer.h"
+#include "env.h"
+#include <openssl/evp.h>
+#include <openssl/hkdf.h>
+#include <jni.h>
+
+using namespace AmazonCorrettoCryptoProvider;
+
+// The possible values of digestCode are defined in HkdfRfc5896SecretKeyFactorySpi.java
+static EVP_MD const* digest_code_to_EVP_MD(int digestCode)
+{
+    switch (digestCode) {
+    case 1:
+        return EVP_sha1();
+    case 2:
+        return EVP_sha256();
+    case 3:
+        return EVP_sha384();
+    default:
```

Comment:
Should we be explicit and throw rather than default to 512?

Reply:
Added an explicit check.